### PR TITLE
zephyr: cmake: Map Zephyr's -Oz abstraction to Picolibc's options

### PR DESCRIPTION
**Pull request by jonathonpenix**
[1] adds a new abstraction for the -Oz optimization level to Zephyr. -Oz will more aggressively optimize for size (relative to -Os), so map this new option to Picolibc's PREFER_SIZE_OVER_SPEED option similar to what is done for -Os.

[1] https://github.com/zephyrproject-rtos/zephyr/pull/70511

### CODE REVIEW

File: zephyr/zephyr.cmake [hunk 1]
```diff
@@ -48,6 +48,16 @@ if(CONFIG_PICOLIBC_USE_MODULE)
     endif()
   endfunction()
 
+  function (picolibc_option_any_true name option_list)
```
Comment (jonathonpenix) [inline zephyr/zephyr.cmake:51]: I'm not sure if this is meaningfully nicer than just doing an `if (SIZE_OPTS OR MIN_SIZE_OPTS) ... else ...` but this seemed in line with the more general `picolibc_option*` functions.

I'm a bit new to cmake yet so apologies in advance if there is a more "canonical" way of doing this and/or if a simple if/else would be sufficient--I'm happy to fix this as needed!

**Review by jonathonpenix — COMMENTED** [review]
It is noted in the commit message/description, but please note that this is a "companion" patch with https://github.com/zephyrproject-rtos/zephyr/pull/70511 so this may need to be fixed accordingly if there are requests for name changes, etc.

Please let me know if it'd be better to close and reopen this if/when https://github.com/zephyrproject-rtos/zephyr/pull/70511 is finalized, otherwise I will just try to keep this updated as https://github.com/zephyrproject-rtos/zephyr/pull/70511 is reviewed (and address any review comments you may have here as well of course). Thanks!

File: zephyr/zephyr.cmake (hunk 1 above)
Comment (keith-packard) [reply zephyr/zephyr.cmake:51]: whatever works; this seems fine to me. cmake is not something I know well.

**Comment by keith-packard** [comment]
I've marked this as a draft until the Zephyr change is finalized. Once that's ready, please switch this back and we can get it merged. Do we need this for 3.7.0?

**Comment by jonathonpenix** [comment]
> I've marked this as a draft until the Zephyr change is finalized. Once that's ready, please switch this back and we can get it merged.

Sounds good, thank you! Will do.

>  Do we need this for 3.7.0?

Er, I personally don't need this in by any particular release--not sure if others feel differently though.

**Pull request merged**